runtime/Makefile

```diff
@@ -27,7 +28,11 @@ $(BCLIB): $(BCS)
 	$(LLVM_LINK) -o - -S $^ | sed -e '/define .*@_[Bb][a-zA-Z]/s/^define /define linkonce_odr /' -e '/target datalayout/ s/"$$/-ni:1"/' | $(LLVM_AS) >$@
 
 %.o: %.c
-	$(CLANG) $(CFLAGS) -c -o $@ $<
+	$(CLANG) -c -O2 -emit-llvm -o - $< | \
+	$(LLVM_LINK) tagged_ptr.ll - -o - -S | \
+	$(OPT) -O2 - -o - -S | \
```

Comment:
Why do you need opt separate from llc?

Reply:
I don't understand. Won't llc do all the necessary optimizations? Why do we need to do "target-independent optimization" separately?

Reply:
As I understood it is like this. clang -O2 does all target-independent and target-dependent optimizations. opt -O2 does target-independent optimizations and llc -O2 does target-dependant optimization. 

If we run `clang-11 -O2 -mllvm --debug-pass=Arguments test.c` we can print all passes that were executed. Then if we run `opt-11 -O2 --debug-pass=Arguments test.ll -S` and `llc-11 -O2 --debug-pass=Arguments test.bc` separately we can see passes executed for those commands, and this output is mostly the same to with clang command output.

In this PR, since I removed clang -O2, I added both opt -O2 and llc -O2 to replace clang -O2.

https://groups.google.com/g/llvm-dev/c/b1ckC_hini4/m/rAkhV18bAQAJ, this is not the exact source I learned this.

Reply:
IIRC clang can accept .ll or .bc as input, so maybe the simplest thing is to use clang again instead of opt/llc (i.e. clang -emit-llvm then llvm-link then clang again)

Reply:
To enable GC for runtime in #300, we have to run statepoint generation pass with opt. Therefore shall we keep it like this?

Reply:
That's a good point. Clang has an optimization pipeline that is carefully crafted to give good results with C, so it woukd be good to take advantage of that. Maybe a combo of opt + clang would work?

Reply:
Yes, changed it.